uli_init/simulate.py

```diff
@@ -704,40 +700,41 @@ def build_molecule(molecule, length, para_weight):
     sequence : list
         List of the configuration sequence of the finished compound
     """
-    f = open("{}/{}.json".format(COMPOUND_DIR, molecule))
+    f = open(f"{COMPOUND_DIR}/{molecule}.json")
     mol_dict = json.load(f)
     f.close()
+
     monomer_sequence = random_sequence(para_weight, length)
-    molecule_string = "{}"
-
-    for idx, config in enumerate(monomer_sequence):
-        if idx == 0:  # append template, but not brackets
-            monomer_string = mol_dict["{}_template".format(config)]
-            if (molecule == "PEEK"):
-                # Change oxygen type on the terminal end of the polymer;
-                # needs its hydrogen.
-                monomer_string = "O" + monomer_string[1:]
-            molecule_string = molecule_string.format(monomer_string)
-            if len(monomer_sequence) == 1:
-                molecule_string = molecule_string.replace("{}", "")
-                continue
-
-        elif idx == length - 1:  # Don't use template for last iteration
-            brackets = polysmiles.count_brackets(
-                mol_dict["{}_deep_smiles".format(config)]
+    compound = Polymer()
+    para = mb.load(mol_dict["para_smiles"], smiles=True, backend="rdkit")
+    meta = mb.load(mol_dict["meta_smiles"], smiles=True, backend="rdkit")
+
+    if len(set(monomer_sequence)) == 2: # Copolymer
+        compound.add_monomer(meta, 
+                mol_dict["meta_bond_indices"],
+                mol_dict["bond_distance"],
+                replace=True
             )
-            monomer_string = mol_dict["{}_deep_smiles".format(config)]
-            molecule_string = molecule_string.format(monomer_string, brackets)
-
-        else:  # Continue using template plus brackets
-            brackets = polysmiles.count_brackets(
-                mol_dict["{}_deep_smiles".format(config)]
+        compound.add_monomer(para,
+                mol_dict["para_bond_indices"],
+                mol_dict["bond_distance"],
+                replace=True
             )
-            monomer_string = mol_dict["{}_template".format(config)]
-            molecule_string = molecule_string.format(monomer_string, brackets)
+    else:
+        if monomer_sequence[0] == "P": # Only para
+            compound.add_monomer(para,
+                    mol_dict["para_bond_indices"],
+                    mol_dict["bond_distance"],
+                    replace=True
+                )
+        elif monomer_sequence[0] == "M": # Only meta
+            compound.add_monomer(meta,
+                    mol_dict["meta_bond_indices"],
+                    mol_dict["bond_distance"],
+                    replace=True
+                )
 
-    molecule_string_smiles = smiles_utils.convert_smiles(deep=molecule_string)
-    compound = mb.load(molecule_string_smiles, smiles=True)
+    compound.build(n=1, sequence=monomer_sequence, add_hydrogens=True)
```

Comment:
OK, I was confused for a minute why n=1 but I think I figured it out c:

Reply:
Yep, in this case, the length of `sequence` is equal to the number of monomers in the molecule.